samcli/lib/package/regexpr.py

```diff
@@ -0,0 +1,4 @@
+"""
+Regular Expressions for Resources.
+"""
+ECR_URL = "(^[a-zA-Z0-9][a-zA-Z0-9-_]*).dkr.ecr.([a-zA-Z0-9][a-zA-Z0-9-_]*).amazonaws.com(.cn)?/.*"
```

Comment:
Should the dots be escaped like this "\."? Otherwise you are matching any char.

Reply:
That makes sense!
```
"(^[a-zA-Z0-9][a-zA-Z0-9-_]*)\.dkr\.ecr\.([a-zA-Z0-9][a-zA-Z0-9-_]*)\.amazonaws\.com(\.cn)?\/.*"
```

is what i tested now and that seems to be more exact.